### PR TITLE
Updated Alertmanager Critical, Default, and Watchdog Receiver InfoTips

### PR DESCRIPTION
**Pull request by dtaylor113**
Updated InfoTips for Alertmanager Receiver configuration.
[Design Doc](https://github.com/cshinn/openshift-origin-design/blob/ce1cb71550fa098330e029a84df94280a77c8d0f/designs/administrator/4.5/alertmanager-config/index.md)

### Receivers Table InfoTip
**_Configure Multiple Receivers_**
image

**_Configure Single Receiver_**
image
**_All Required Receivers Configured (no InfoTip)_**
image

### Default Receiver InfoTip ###
image
**_InfoTip remains after Default Receiver configured_** 
image

### Critical Receiver InfoTip
image
**_InfoTip not shown after receiver configured_**
image

### Watchdog Receiver InfoTip
image
**_InfoTip not shown after receiver configured_**
image

JIRA: https://issues.redhat.com/browse/CONSOLE-2055

### CODE REVIEW

**Comment by cshinn** [comment]
I just realized that the copy for the default receiver doesn't really make sense now that we provide multiple receivers at the same time; there isn't really a "first" one anymore. I'll work on some updated text.

**Comment by dtaylor113** [comment]
/hold

Waiting for new verbiage from @cshinn :-)

**Comment by dtaylor113** [comment]
> ...I'll work on some updated text.

Thanks @cshinn

**Comment by dtaylor113** [comment]
Hi @cshinn, wondering if Watchdog should end with "...this receiver should remain in its current state _with no Receiver Type selected_."  -or something to that extent  -thanks

**Comment by cshinn** [comment]
@dtaylor113 
Default Receiver:
Your default receiver will automatically receive all alerts from this cluster that are not caught by other receivers first. 

Watchdog:
The Watchdog alert fires constantly to confirm that your alerting stack is functioning correctly. This receiver is configured to prevent it from creating unnecessary notifications. You can edit this receiver if you plan to use the information that Watchdog provides, otherwise this receiver should remain in its current state with no set receiver type.

**Comment by dtaylor113** [comment]
@cshinn, thanks for the updates.  Latest screenshots:
image
image

File: frontend/public/components/monitoring/alert-manager-config.tsx [hunk 1]
```diff
@@ -279,7 +286,7 @@ const ReceiverTableRow: React.FC<ReceiverTableRowProps> = ({
     <TableRow id={index} index={index} trKey={key} style={style}>
       <TableData className={tableColumnClasses[0]}>{receiver.name}</TableData>
       <TableData className={tableColumnClasses[1]}>
-        {!integrationTypesLabel ? (
+        {(receiver.name === 'Critical' || receiver.name === 'Default') && !integrationTypesLabel ? (
```
Comment (benjaminapetersen) [inline frontend/public/components/monitoring/alert-manager-config.tsx]: I see `.name === 'Critical'` several times across these files.  Do you want to extract into a `isCritical(thingWithName)` func, and then make the `Critical` and `Default` an enum?

File: frontend/public/components/monitoring/alert-manager-config.tsx [hunk 2]
```diff
@@ -175,11 +175,18 @@ const hasSimpleReceiver = (
   return false;
 };
 
-const hasIncompleteDefaultReceiver = () => {
+const numberOfIncompleteReceivers = (): number => {
```
Comment (benjaminapetersen) [inline frontend/public/components/monitoring/alert-manager-config.tsx]: Not new, just wondering why `config` isn't a func arg.

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx [hunk 3]
```diff
@@ -173,6 +173,37 @@ const getRouteLabelsForEditor = (
     : routeLabels;
 };
 
+const ReceiverInfoTip: React.FC<ReceiverInfoTipProps> = ({ type }) => {
+  let msg: string = '';
```
Comment (benjaminapetersen) [inline frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: How about dropping the `let` and the `break` statements:

```
case 'Default':
   return 'Your default receiver will automatically receive al.....'
case 'Critical':
   return 'The routing labels for this receiv....'
case 'Watchdog':
  return 'The Watchdog alert fires consta....'
```

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx [hunk 4]
```diff
@@ -580,3 +604,7 @@ type SaveAsDefaultCheckboxProps = {
   dispatchFormChange: Function;
   tooltip: string;
 };
+
+type ReceiverInfoTipProps = {
```
Comment (benjaminapetersen) [inline frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: Seems like you may want to make an `enum` of these 3 values and use it throughout.

File: frontend/public/components/monitoring/alert-manager-config.tsx [hunk 5]
```diff
@@ -175,11 +175,18 @@ const hasSimpleReceiver = (
...
   const { route, receivers } = config;
   const { receiver: defaultReceiverName } = route;
-  const defaultReceiver = _.filter(receivers, { name: defaultReceiverName });
-  return defaultReceiver.length === 1 && _.isEmpty(getIntegrationTypes(defaultReceiver[0]));
+
+  const defaultReceiver = receivers.filter((receiver) => receiver.name === defaultReceiverName);
+  const criticalReceiver = receivers.filter((receiver) => receiver.name === 'Critical');
+
```
Comment (benjaminapetersen) [inline frontend/public/components/monitoring/alert-manager-config.tsx]: Possible values then are `0,1,2`?

Comment (dtaylor113) [reply frontend/public/components/monitoring/alert-manager-config.tsx]: yes

File: frontend/public/components/monitoring/alert-manager-config.tsx (hunk 1 above)
Comment (dtaylor113) [reply frontend/public/components/monitoring/alert-manager-config.tsx]: Added InitialReceivers enum.  Not sure `isCritical()` is needed as the conditional seems easier to read to me, is only used in two places, and matches how we test for Default initial receiver.

File: frontend/public/components/monitoring/alert-manager-config.tsx (hunk 2 above)
Comment (dtaylor113) [reply frontend/public/components/monitoring/alert-manager-config.tsx]: `config` started out as a func arg but I ended up passing it to virtually every function so seemed logical to make it a global to be used throughout.

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx (hunk 3 above)
Comment (dtaylor113) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: Nice, much cleaner -thanks

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx (hunk 4 above)
Comment (dtaylor113) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: fixed

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx [hunk 6]
```diff
@@ -580,3 +600,7 @@ type SaveAsDefaultCheckboxProps = {
...
+  type: InitialReceivers.Watchdog | InitialReceivers.Critical | InitialReceivers.Default;
```
Comment (TheRealJon) [inline frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: Shouldn't this just be:

```suggestion
  type: InitialReceivers;
```

Comment (dtaylor113) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: yup, thanks, fixed :-)

**Pull request merged**